src/desktop/camera.rs

```diff
@@ -115,81 +115,161 @@ impl<'a> CameraProxy<'a> {
     }
 }
 
-/// A helper to get the PipeWire Node ID to use with the camera file descriptor
-/// returned by [`CameraProxy::open_pipe_wire_remote`].
+#[cfg(feature = "pipewire")]
+fn foreign_dic_to_map<D: pw::prelude::ReadableDict>(foreign: &D) -> HashMap<String, String> {
+    let mut map = HashMap::new();
+    for (key, value) in foreign.iter() {
+        map.insert(key.to_string(), value.to_string());
+    }
+    map
+}
+
+#[cfg(feature = "pipewire")]
+/// A PipeWire camera stream returned by [`pipewire_streams`].
+pub struct CameraStream {
+    node_id: u32,
+    properties: HashMap<String, String>,
+}
+
+#[cfg(feature = "pipewire")]
+impl CameraStream {
+    /// The id of the PipeWire node.
+    pub fn node_id(&self) -> u32 {
+        self.node_id
+    }
+
+    /// Map of node properties.
+    pub fn properties(&self) -> HashMap<String, String> {
+        self.properties.clone()
+    }
+
+    /// The nick of the PipeWire node.
+    pub fn nick(&self) -> Option<String> {
+        self.properties.get("nick").cloned()
+    }
+
+    /// The name of the PipeWire node.
+    pub fn name(&self) -> Option<String> {
+        self.properties.get("name").cloned()
+    }
+
+    /// The description of the PipeWire node.
+    pub fn description(&self) -> Option<String> {
```

Comment:
I am a bit hesitant about providing those getters here tbh, probably worth waiting until we have confirmation about what is common and what we can assume always exists & what camera users would expect to have easily retrievable

Reply:
In practice these are the 3 most important properties one cares about, on the 2 devices I tried they were all present.

Reply:
sure, the thing is adding new apis in the future is easier than dropping it :P